aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -94,6 +94,42 @@ void div_kernel(TensorIterator& iter) {
   }
 }
 
+void remainder_kernel(TensorIterator& iter) {
+  if (isIntegralType(iter.dtype(), /*includeBool*/ false)) {
+    AT_DISPATCH_INTEGRAL_TYPES(iter.dtype(), "remainder_cpu", [&]() {
+      cpu_kernel(iter, [](scalar_t a, scalar_t b) -> scalar_t {
+        TORCH_CHECK(b != 0, "ZeroDivisionError");
+        scalar_t r = a % b;
+        if ((a != 0) && (a < 0) != (b < 0)) {
+          r += b;
+        }
+        return r;
+      });
+    });
+  } else {
+    AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.dtype(), "remainder_cpu", [&]() {
+      cpu_kernel_vec(iter,
+        [=](scalar_t a, scalar_t b) __ubsan_ignore_float_divide_by_zero__ -> scalar_t {
+          if (b == 0) {
+            return std::numeric_limits<scalar_t>::quiet_NaN();
+          } else {
+            return a - b * at::native::floor_impl(a / b);
+          }
+        },
+        [=](Vec256<scalar_t> a, Vec256<scalar_t> b) {
+          Vec256<scalar_t> r = a - b * (a / b).floor();
+          Vec256<scalar_t> r_nans = b.map([](const scalar_t x) -> scalar_t {
```

Comment:
I actually don't know if this part is necessary -- I couldn't find any documentation that defines precisely what the intrinsic does.  And this is certainly going to be slower, unlikely the non-vectorized case.

I was just suggesting adding a test for the vectorized path and seeing if the ubsan / asan build complains.  If it doesn't I think the old vectorized code is fine (might want to add a comment though).

Reply:
Oh alright, makes sense. I'll add a test and see what happens.

Reply:
The test cases I added for this pass whether or not the explicit divide by 0 check is there, so I removed the check.